Assets/Scripts/UnitController.cs

```diff
@@ -59,6 +63,29 @@ void FixedUpdate()
 
     void UnitMovement()
     {
+        if (IsOffScreenVertically(border1Sprite) && IsOffScreenVertically(border2Sprite) && IsOffScreenVertically(border3Sprite) && IsOffScreenVertically(border4Sprite))
+        {
+            if (whitePoint == null) whitePoint = Instantiate(whiteCirclePrefab);
+
+            SpriteRenderer whiteSprite = whitePoint.transform.Find("Sprite").GetComponent<SpriteRenderer>();
```

Comment:
This is a performance issue, again.
Less problematic than in the previous PR, absolutely. But you are still fetching a sprite of an object every physics update.

When you need a reference to something often, please make it so that you only have to fetch it once.